src/main/java/core/basesyntax/ArrayList.java

```diff
@@ -1,48 +1,146 @@
 package core.basesyntax;
 
+import java.util.NoSuchElementException;
+
 public class ArrayList<T> implements List<T> {
+    private int arraySize;
+    private Object[] elementData;
+    private int size;
+
+    public ArrayList() {
+        arraySize = 10;
+        elementData = new Object[arraySize];
+        size = 0;
+    }
+
     @Override
     public void add(T value) {
-
+        if (size >= arraySize) {
+            Object[] obj = elementData;
+            elementData = new Object[(int) (elementData.length + (elementData.length * 0.5))];
+            for (int i = 0; i < obj.length; i++) {
+                elementData[i] = obj[i];
+            }
+            arraySize = elementData.length;
```

Comment:
The logic for growing the array is repeated in several methods. Consider extracting it into a separate private method to adhere to the DRY principle. Also, use System.arraycopy() to move your array elements and resize the array in a separate method according to the Single Responsibility principle.